send-poke

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+# Send an "I am alive" ping to Canonical. This is used for surveying how many
+# original OEM installs are still existing on real machines. Note that this
+# does not send any user specific data; it only transmits the operating system
+# version (/var/lib/ubuntu_dist_channel), the machine product name, and a
+# counter how many pings were sent.
+#
+# (C) 2014 Canonical Ltd.
+# License: GPL v2 or later
+
+set -e
+
+# We need to know if we are being run by cron or manually.
+PERIOD=$(echo $(dirname $0) | sed -e 's/.*\.//' | \
+    sed -n '/hourly\|daily\|weekly\|monthly/ p')
+PERIOD=${PERIOD:-manual}
+
+COUNTDIR=/var/lib/send-install-count
+FIRSTRUNFILE=$COUNTDIR/firstrun
+# We want to send a different count depending on the determined period
+if [ $PERIOD == daily ]; then
+    COUNTFILE=$COUNTDIR/counter
+else
+    COUNTFILE=$COUNTDIR/counter-$PERIOD
+fi
```

Comment:
leaving this file verbatim from canonical-poke for now, it will be rewritten into a "ping" script analagous to "activate" if we decide to pursue that approach